app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/survey/SurveyDetailViewModel.kt

```diff
@@ -23,4 +42,32 @@ class SurveyDetailViewModel @Inject constructor() : ViewModel() {
     fun setCurrentPage(pageNumber: Int) {
         _currentPage.value = pageNumber
     }
+
+    fun getSurveyQuestions(surveyId: String) {
+        viewModelScope.launch(ioDispatcher) {
+            surveyRepo.getSurveyDetails(surveyId).collect {
+                when (it) {
+                    is ResourceState.Loading -> {
+                        _shouldShowLoading.value = true
+                        resetError()
+                    }
+                    is ResourceState.Success -> {
+                        _shouldShowLoading.value = false
+                        _surveyQuestions.value = it.data
+                        resetError()
+                    }
+                    else -> {
```

Comment:
[Optional] We could be more explicit here just in case we add any more `ResourceState`s, but no biggie since we most likely won't add any 😄 
```suggestion
                    is ResourceState.Error,
                    is ResourceState.NetworkError -> {
```